src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.D.cs

```diff
@@ -194,144 +196,128 @@ private static bool TryParseInt16D(ReadOnlySpan<byte> source, out short value, o
-                if (!ParserHelpers.IsDigit(num))
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
-                    goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
-                    goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
-                    goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
-                    goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
-                    goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
-                if ((uint)index >= (uint)source.Length)
+                if (!ParserHelpers.TryGetDigitAt(source, index, out num))
                     goto Done;
-                num = source[index];
-                if (!ParserHelpers.IsDigit(num))
-                    goto Done;
-                index++;
-                answer = 10 * answer + num - '0';
+                index += 1;
+                answer = 10 * answer + num;
 
                 // Potential overflow
```

Comment:
The comment is obsolete now.

Reply:
I don't think so. We still check for overflow in the lines which immediately follow.